### DISCUSSION TITLE
Python scripts included

### DISCUSSION THREAD

**Discussion by martinmiglio** [marcelotduarte/cx_Freeze#1855] [Q&A]
In my project, I package it into an MSI using cx_freeze, but the project has a dependency on ahk[binary], which installs a binary file, `AutoHotKey.exe` to the `Scripts/` dir in my virtual environment, along side `python.exe` and other compiled scripts. 
Is it possible to include this file (or directory) with out hard coding it's path in the build script? Is there an option to do this as part of the build process?

**Comment by marcelotduarte** [comment]
Use [include_files](https://cx-freeze.readthedocs.io/en/latest/setup_script.html#cmdoption-arg-include_files)

I also suggest you to use AHK(executable_path='C:\\path\\to\\AutoHotkey.exe') as documented [here](https://github.com/spyoungtech/ahk#non-python-dependencies)

So, in setup, use "include_files":[r"C:\...\AutoHotkey.exe"] to copy the file.
And in your main script:
```python
if getattr(sys, "frozen", False):
    # The application is frozen
    ahk = AHK(executable_path=os.path.join(sys.frozen_dir, "AutoHotkey.exe"))
else:
    ahk = AHK()
```

**Reply by martinmiglio** [reply]
I knew the solution was using `include_files`, but I was looking for an elegant solution to include the AHK binary which is installed from AHK[binary] as a dependency in the current environment's script folder, regardless of where that may be. I ended up doing something with poetry (my package and environment manager) to find the exe for me, agnostic of the environment. It looks something like this:
```python
# use poetry to determine path to interpreter
poetry_factory = poetry.factory.Factory()
pt = poetry_factory.create_poetry()
venvs_path = Path(pt.config.get("virtualenvs.path"))

# select the first venv path starting with project name and ending with python version
# (e.g. mypackagename-XXXXXXXX-3.9)
python_version = f"{sys.version_info.major}.{sys.version_info.minor}"
venv = next(
    venv
    for venv in venvs_path.iterdir()
    if venv.name.startswith(pt.package.name) and venv.name.endswith(python_version)
)

# find AutoHotKey.exe in Scripts folder
scripts_path = venv / "Scripts"
ahk_path = next(scripts_path.glob("AutoHotKey*.exe"))

build_exe_options = {
    "include_files": [ahk_path],
}
```

**Reply by marcelotduarte** [reply]
Ah... sys.prefix

**Reply by martinmiglio** [reply]
> Ah... sys.prefix

don't tell me its that simple 😅

**Reply by martinmiglio** [reply]
```python
scripts_path = Path(sys.prefix) / "Scripts"
ahk_path = next(scripts_path.glob("AutoHotKey*.exe"))
```

so much simpler...
Thanks!
